### PR TITLE
Fix HTTPStore prepare multi-fetch

### PR DESCRIPTION
**Pull request by mrtbld**
HTTPStore prepare function fetched http stores regardless of whether
they have already been successfully fetched (i.e. resolved status) or
not.

It now checks whether the store is resolved or rejected before
fetching it, in which case it doesn't do anything.

### CODE REVIEW

File: lib/stores.js [hunk 1]
```diff
@@ -29,7 +29,14 @@ function stores(
   return (Component) => preparable(async function prepare(props, context) {
     const flux = context[fluxKey];
     const bindings = getBindings(props, flux);
-    const deps = _.map(bindings, (binding) => flux.fetchStore(binding));
+    const deps = _.map(bindings, (binding) => {
+      const state = flux.readStoreFromState(binding);
+      if(state.isResolved()) {
```
Comment (mrtbld) [inline lib/stores.js]: Should handle isRejected the same way. (But not uninitialized.)

File: lib/stores.js [hunk 2]
```diff
@@ -29,7 +29,14 @@ function stores(
   return (Component) => preparable(async function prepare(props, context) {
     const flux = context[fluxKey];
     const bindings = getBindings(props, flux);
-    const deps = _.map(bindings, (binding) => flux.fetchStore(binding));
+    const deps = _.map(bindings, (binding) => {
+      const state = flux.readStoreFromState(binding);
+      if(state.isResolved() || (state.isRejected() && !state.meta.uninitialized)) {
```
Comment (antca) [inline lib/stores.js]: An explicit check of the existence of `state.meta` is required i think.

File: lib/__tests__/node/prepare.js [hunk 3]
```diff
@@ -80,4 +87,58 @@ describe('prepare', () => {
...
   });
+
+  it('doesn’t fetch the same HTTPStore more than once', async function test() {
+    const testHttpConf = {
+      protocol: 'http',
+      hostname: 'test.example.local',
+      port: '42',
+    };
+    const testStorePath = '/testHttpStore';
+    const apiMock = nock(_.template('${protocol}://${hostname}:${port}')(testHttpConf))
```
Comment (antca) [inline lib/__tests__/node/prepare.js]: I think `url.format` is better il this case: `nock(url.format(testHttpConf))`.

File: lib/stores.js (hunk 2 above)
Comment (mrtbld) [inline lib/stores.js]: Yup :+1:

Comment (mrtbld) [inline lib/stores.js]: I changed the State class so that meta defaults to empty object: b05b8fe97ac936e57b5658309f054835605c9b28

OK for you?

Comment (antca) [inline lib/stores.js]: :+1:

**Pull request merged**